From: Felicia Lim <flim@google.com>
Date: Tue, 3 Dec 2019 00:37:14 +0000 (-0800)
Subject: Don't update null data pointer after each multistream decoder
X-Git-Url: http://git.xiph.org/?p=opus.git;a=commitdiff_plain;h=8493bf50892be0e902a354444747e6d14f2a6a10

Don't update null data pointer after each multistream decoder

The data pointer could be null in the case of DTX or packet loss.
---

diff --git a/src/opus_multistream_decoder.c b/src/opus_multistream_decoder.c
index 0018517a..a2837c35 100644
--- a/src/opus_multistream_decoder.c
+++ b/src/opus_multistream_decoder.c
@@ -251,8 +251,11 @@ int opus_multistream_decode_native(
       }
       packet_offset = 0;
       ret = opus_decode_native(dec, data, len, buf, frame_size, decode_fec, s!=st->layout.nb_streams-1, &packet_offset, soft_clip);
-      data += packet_offset;
-      len -= packet_offset;
+      if (!do_plc)
+      {
+        data += packet_offset;
+        len -= packet_offset;
+      }
       if (ret <= 0)
       {
          RESTORE_STACK;
